pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/steps/CredentialWrapperStepTest.java

```diff
@@ -0,0 +1,123 @@
+
+        StringCredentialsImpl mixedEnvCred1 = new StringCredentialsImpl(CredentialsScope.GLOBAL, mixedEnvCred1Id, "test", Secret.fromString(mixedEnvCred1Secret));
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), mixedEnvCred1);
+        UsernamePasswordCredentialsImpl mixedEnvCred2 = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, mixedEnvCred2Id, "sample", mixedEnvCred2U, mixedEnvCred2P);
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), mixedEnvCred2);
+
+        folder = j.jenkins.createProject(Folder.class, "testFolder");
+        StringCredentialsImpl sc = new StringCredentialsImpl(CredentialsScope.GLOBAL, mixedEnvCred1Id, "test", Secret.fromString(mixedEnvInFolderCred1Secret));
+        CredentialsProvider.lookupStores(folder).iterator().next().addCredentials(Domain.global(), sc);
+        UsernamePasswordCredentialsImpl c = new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL, mixedEnvCred2Id, "sample", mixedEnvInFoldercred2U, mixedEnvInFolderCred2P);
+        CredentialsProvider.lookupStores(folder).iterator().next().addCredentials(Domain.global(), c);
+
+        SSHUserPrivateKey k = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, "sshCred1", "bobby", new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource("abc123"), null, "sample");
+        CredentialsProvider.lookupStores(j.jenkins).iterator().next().addCredentials(Domain.global(), k);
+    }
+
+
+    @Test
+    public void usernamePassword() throws Exception {
+        expect("credentials", "usernamePassword").runFromRepo(false)
+                .logNotContains(usernamePasswordPassword, "FOO_USR is " + usernamePasswordUsername)
+                .logContains("FOO_USR is *")
+                .archives("combined/foo.txt", allOf(containsString(usernamePasswordUsername), containsString(usernamePasswordPassword)))
+                .archives("foo_usr.txt", usernamePasswordUsername).archives("foo_psw.txt", usernamePasswordPassword).go();
+    }
+
+    @Test
+    public void mixedEnv() throws Exception {
+        expect("credentials", "mixedEnv").runFromRepo(false)
+                .logContains("SOME_VAR is SOME VALUE",
+                             "INBETWEEN is Something in between",
+                             "OTHER_VAR is OTHER VALUE")
+                .archives("cred1.txt", mixedEnvCred1Secret)
+                .archives("cred2.txt", mixedEnvCred2U + ":" + mixedEnvCred2P).go();
+    }
+
+    @Test
+    public void mixedEnvInFolder() throws Exception {
+
+        expect("credentials", "mixedEnv").runFromRepo(false).inFolder(folder)
+                .logContains("SOME_VAR is SOME VALUE",
+                             "INBETWEEN is Something in between",
+                             "OTHER_VAR is OTHER VALUE")
+                .archives("cred1.txt", mixedEnvInFolderCred1Secret)
```

Comment:
Turned out that a good old fashioned `configRoundTrip` was needed and then just fetch the store from the property directly.